gobblin-compaction/src/main/java/gobblin/compaction/dataset/Dataset.java

```diff
@@ -275,15 +272,27 @@ public void resetNeedToRecompact() {
     this.needToRecompact = false;
   }
 
+  private void cleanAdditionalInputPath () {
+    this.additionalInputPaths.clear();
+  }
+
+
   /**
    * Modify an existing dataset to recompact from its ouput path.
    */
   public void modifyDatasetForRecompact(State recompactState) {
-    this.setJobProps(recompactState);
-    this.setInputPath(this.outputPath);
-    this.setInputLatePath(this.outputLatePath);
-    this.addAdditionalInputPath(this.outputLatePath);
-    this.resetNeedToRecompact();
+    if (!this.jobProps().getPropAsBoolean(MRCompactor.COMPACTION_RECOMPACT_ALL_DATA, MRCompactor.DEFAULT_COMPACTION_RECOMPACT_ALL_DATA)) {
+      this.setJobProps(recompactState);
+      this.setInputPath(this.outputLatePath);
+      this.cleanAdditionalInputPath();
+      this.resetNeedToRecompact();
+    } else {
```

Comment:
Most of the code in both conditional branches are same. Can you remove them out of the `if`/`else`?